packages/datadog-plugin-langchain/src/handlers/default.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+const Sampler = require('../../../dd-trace/src/sampler')
+
+const RE_NEWLINE = /\n/g
+const RE_TAB = /\t/g
+
+// TODO: should probably refactor the OpenAI integration to use a shared LLMTracingPlugin base class
+// This logic isn't particular to LangChain
```

Comment:
I will create a ticket for this, and will probably try and refactor all LLM _tracing_ plugins (openai, langchain, and then also anthropic, maybe AWS bedrock if possible) after they land. I tried to see how trivial it would be to include in this PR, but it's pretty non-trivial, since the OpenAI tracing plugin is so massive.

Reply:
I wonder if we can do this work after the openai and langchain PRs land before anthropic/bedrock are in progress. Otherwise if it's too much work, definitely fine to do this in a later PR, but let's make sure we track this work!

Reply:
Yes I think that should be doable! I'll make a ticket regardless but I agree it should be done sooner rather than later